source/developer/webapp-component.rst

```diff
@@ -0,0 +1,180 @@
+
+- https://facebook.github.io/react/tutorial/tutorial.html
+- http://redux.js.org/
+
+The components for the webapp currently live in the ``webapp/components/`` directory in the `platform repository <https://github.com/mattermost/platform>`__.
+
+Designing your Component
+------------------
+
+The most important part of designing your component is decidiing on what the props will be. Props are very much the API for your component, think of them as a contract between your component and the users of it.
+
+Props are read-only variables that get passed down to your component either directly from a parent component or from an `index.js` container that pulls some data from the Redux store. More on that later, for now you can see all the different prop types here: https://facebook.github.io/react/docs/typechecking-with-proptypes.html#proptypes.
+
+Alright, so how do you decide what props your component should have? Think about your component is trying to display to the user. Any data you need to accomplish that should be part of the props.
+
+As an example, let's imagine we're building an ItemList component with the purpose of displaying a list of items. The props for such a component might look like:
+
+.. code-block:: javascript
+  :linenos:
+
+  static propTypes = {
+      /**
+       * Sets the title of the list
+       */
+      title: PropTypes.string,
+
+      /**
+       * An array of item components to display
+       */
+      items: PropTypes.arrayOf(PropTypes.object).isRequired
+  }
+
+The ``title`` prop would control the string to display for the title, while ``items`` would be an array of the item objects we would like to display. Note that ``items`` has ``.isRequired`` appended while ``title`` does not. This means that to render our component must have the ``items`` prop set, while it can still render if we don't set ``title``. Use this to provide optional props.
+
+Make sure you add brief but clear comments to each prop type as shown in the example.
+
+Our ItemList component would live in a file named ``item_list.js``.
+
+Using a Container
+------------------
+
+The next question to ask yourself is if you're going to need a container for your component. This is the ``index.js`` file mentioned above. If your component needs either of the following then you'll need a container:
+
+1. Needs some data injected into it's props that the parent component doesn't have access to
+2. Needs to be able to perform some sort of action that affects the state of the store
+
+Continuing the ItemList example above, maybe our parent component doesn't care about our list of items and doesn't have access to them. Let's also imagine that we want to let the user remove items from the list by clicking on them. This means our component now needs a container for both criteria above and our props will change slightly:
+
+.. code-block:: javascript
+  :linenos:
+
+  static propTypes = {
+      /**
+       * Sets the title of the list
```

Comment:
What about using either single line comments or just `/*block comments like this*/` instead of having these on multiple lines? I don't like the extra vertical space used just for the  start and end of the comments